idb/postgres/postgres_migrations.go

```diff
@@ -342,70 +382,271 @@ func m5accountCumulativeRewardsUpdate(db *IndexerDb, state *MigrationState) erro
 	// Get the remainder
 	if len(accounts) > 0 {
 		db.log.Println("Processing final batch of accounts.")
-		m5accountCumulativeRewardsUpdateAccounts(db, state, accounts, true)
+		err := m5RewardsAndDatesPart2UpdateAccounts(db, state, accounts, true)
+		if err != nil {
+			return err
+		}
 		accounts = accounts[:0]
 	}
 
 	return nil
 }
 
-// m5accountCumulativeRewardsUpdateAccounts loops through the provided accounts and generates a bunch of updates in a
-// single transactional commit.
-func m5accountCumulativeRewardsUpdateAccounts(db *IndexerDb, state *MigrationState, accounts []string, finalBatch bool) error {
+type createClose struct {
+	created sql.NullInt64
+	closed  sql.NullInt64
+}
+
+// updateClose will update the closed round only if it is the first event (going from newest to oldest).
+func updateClose(cc *createClose, value uint64) *createClose {
+	if cc == nil {
+		return &createClose{
+			closed: sql.NullInt64{
+				Valid: true,
+				Int64: int64(value),
+			},
+		}
+	}
+
+	return cc
+}
+
+// updateCreate will update the created round.
+func updateCreate(cc *createClose, value uint64) *createClose {
+	if cc == nil {
+		return &createClose{
+			created: sql.NullInt64{
+				Valid: true,
+				Int64: int64(value),
+			},
+		}
+	}
+
+	cc.created.Valid = true
+	cc.created.Int64 = int64(value)
+
+	return cc
+}
+
+func executeForEachCreatable(stmt *sql.Stmt, address []byte, m map[uint64]*createClose) (err error) {
+	for index, round := range m {
+		_, err = stmt.Exec(address, index, round.created, round.closed)
+		if err != nil {
+			return
+		}
+	}
+	return
+}
+
+// m5RewardsAndDatesPart2UpdateAccounts loops through the provided accounts and generates a bunch of updates in a
+// single transactional commit. These queries are written so that they can run in the background.
+//
+// For each account we run several queries:
+// 1. setTotalRewards            - conditionally set the total rewards if the account wasn't closed during iteration.
+// 2. setCreateCloseAccount      - set the accounts create/close rounds.
+// 3. setCreateCloseAsset        - set the accounts created assets create/close rounds.
+// 4. setCreateCloseAssetHolding - (upsert) set the accounts asset holding create/close rounds.
+// 5. setCreateCloseApp          - set the accounts created apps create/close rounds.
+// 6. setCreateCloseAppLocal     - (upsert) set the accounts local apps create/close rounds.
```

Comment:
@brianolson if you could take a close look at these queries that would be really helpful. I have migration tests in place, but there's no activity during the test so they don't test the "non-blocking" part.